remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/JsonRemoteMessageMapper.kt

```diff
@@ -100,11 +99,17 @@ private val defaultBrowserActionMapper: (JsonMessageAction) -> Action = {
     Action.DefaultBrowser()
 }
 
+private val appTpOnboardingActionMapper: (JsonMessageAction) -> Action = {
+    Action.AppTpOnboarding()
+}
+
+// plugin point?
```

Comment:
The comment was more to come back to it.
In principle all mappers could be plugins, that would allow mappers to be added from other modules.

The main issue with mappers right now, is that the whole logic is govern by higher order extension functions, and turning them into plugins requires big(ger) refactor.

Reply:
Yeah, the `actionMappers` should be plugin...eventually. There's not reason for the remote-message feature to know anything about `APP_TP_ONBOARDING` action.
But as said below, it's harder to turn this into plugins right now, because they're called higher order functions and extension functions, and need a big(ger) refactor.

So left this as a note to come back to it later.

Reply:
sounds good